src/core/vector/qgsvectorlayertemporalproperties.cpp

```diff
@@ -443,22 +443,20 @@ QString QgsVectorLayerTemporalProperties::createFilterString( const QgsVectorLay
     {
       if ( mAccumulateFeatures )
       {
-        return QStringLiteral( "(%1 %2 %3) OR %1 IS NULL" ).arg( QgsExpression::quotedColumnRef( mStartFieldName ),
```

Comment:
we need to leave the range.includeEnd() ? QStringLiteral( "<=" ) : QStringLiteral( "<" ), logic here -- while the gui isn't creating ranges which includes the end, the raw api may still be used to set a map range with inclusive end and it's important it's correctly handled here